packages/core/ai/src/util/printer.ts

```diff
@@ -8,25 +8,40 @@ import type { Message, MessageContentBlock } from '../tools';
 
 type Mode = 'text' | 'json';
 
+// TODO(burdon): Provide structured logger.
+export type Logger = (...data: any[]) => void;
+
+export type ConsolePrinterOptions = {
+  logger?: Logger;
+  mode?: Mode;
+};
+
 export class ConsolePrinter {
+  logger: Logger;
   mode: Mode;
-  constructor({ mode = 'text' }: { mode?: Mode } = {}) {
+
+  // eslint-disable-next-line no-console
+  constructor({ logger = console.log, mode = 'text' }: ConsolePrinterOptions = {}) {
```

Comment:
is there a reason you don't want to use `@dxos/log` here?

Reply:
i'm not sure; refactoring @dmaretskyi; i don't think so.